src/treelearner/monotone_constraints.hpp

```diff
@@ -498,15 +499,15 @@ class BasicLeafConstraints : public LeafConstraintsBase {
     return std::vector<int>();
   }
 
-  const ConstraintEntry* Get(int leaf_idx) override { return entries_[leaf_idx]; }
+  const ConstraintEntry* Get(int leaf_idx) override { return entries_[leaf_idx].get(); }
 
   FeatureConstraint* GetFeatureConstraint(int leaf_idx, int feature_index) final {
     return entries_[leaf_idx]->GetFeatureConstraint(feature_index);
   }
 
  protected:
   int num_leaves_;
-  std::vector<ConstraintEntry*> entries_;
+  std::vector<std::unique_ptr<ConstraintEntry>> entries_;
```

Comment:
cc @CharlesAuguste  for the changes regards to memory leakage in MC.

Reply:
Looks good, sorry about the leakage.